### ISSUE TITLE
Zbay should be performant enough to run all the time in the background, like Slack or Discord

### ISSUE THREAD

**Issue by holmesworcester** [ZbayApp/zbay#463]
I'm subscribed to quite a few channels / viewing keys, but there isn't much activity on any of them. Still, Zbay CPU is hovering around 100% even when I'm not sending messages. 

We'll do some profiling, propose plan to Marek and HW, then move forward.

**Comment by NorbertBodziony** [comment]
I am not experiencing it.

**Comment by holmesworcester** [comment]
@kingalg let's try joining *lots* of public channels and then test performance. Just to see if you can reproduce...

**Comment by holmesworcester** [comment]
Next step if we can reproduce: ask aditya, or discuss.

**Comment by holmesworcester** [comment]
@kingalg can you reproduce any slowdown when you're using lots of channels?

**Comment by holmesworcester** [comment]
Performance improvements in zecwallet-light-cli may have improved this

**Comment by holmesworcester** [comment]
App is still responsive, but CPU is 100% and fans are spinning

**Comment by holmesworcester** [comment]
@NorbertBodziony can you add your performance improvement idea re: saving here?

**Comment by holmesworcester** [comment]
Another option is to not sync every block. (We could sync every 10 blocks)

**Comment by holmesworcester** [comment]
Another option is to check if there's a new block and only do a full fetch when there is a new block. (This will also increase speed with which we retrieve new transactions.)

**Comment by holmesworcester** [comment]
Other ideas are to change the syncing behavior and frequency based on whether the window has focus. Or whether the window is even open

**Comment by holmesworcester** [comment]
We are also checking the signature on every message on every sync instead of just on new messages. We could also see if a message was new before unpacking the memo.

**Comment by holmesworcester** [comment]
We're going to start by reducing CPU usage when we're receiving messages.

https://github.com/ZbayApp/zbay/issues/564

**Comment by holmesworcester** [comment]
@NorbertBodziony after we complete this, what would be the next lowest hanging fruit performance improvement? Reduce the sync frequency to 1 per block?

**Comment by NorbertBodziony** [comment]
Yes this will be the last thing that may be heavy, but we should first check if fixes we already got will be enough.

**Comment by holmesworcester** [comment]
@NorbertBodziony are there any more performance improvements we want to try? or should i just assign this to myself and see how it feels?

**Comment by NorbertBodziony** [comment]
I think you should check it since you will be able to compare it.

**Comment by holmesworcester** [comment]
I'm still seeing pretty consistent 100% CPU in Zbay even when I'm not sending or receiving z messages.
